lib-injection/auto_inject.rb

```diff
@@ -0,0 +1,84 @@
+return if ENV['skip_autoinject']
+
+begin
+  require 'open3'
+
+  _, status = Open3.capture2e({'skip_autoinject' => 'true'}, 'bundle show ddtrace')
+
+  if status.success?
+    puts '[DATADOG LIB INJECTION] ddtrace already installed... skipping auto-injection'
+    return
+  end
+
+  require 'bundler'
+  require 'shellwords'
+
+  if Bundler.frozen_bundle?
+    puts "[DATADOG LIB INJECTION] Cannot inject with frozen Bundler"
+    return
+  end
+
+  version = "<DD_TRACE_VERSION_TO_BE_REPLACED>"
+  sha = "<DD_TRACE_SHA_TO_BE_REPLACED>"
+
+  bundle_add_ddtrace_cmd =
+    if !version.empty?
+      # For public release
+      "bundle add ddtrace --require ddtrace/auto_instrument --version #{version.gsub(/^v/, '').shellescape}"
+    elsif !sha.empty?
+      # For internal testing
+      "bundle add ddtrace --require ddtrace/auto_instrument --github datadog/dd-trace-rb --ref #{sha.shellescape}"
+    end
+
+  unless bundle_add_ddtrace_cmd
+    puts "[DATADOG LIB INJECTION] NO VERSION"
+    return
+  end
+
+  puts "[DATADOG LIB INJECTION] ddtrace is not installed... Perform lib injection for dd-trace-rb."
```

Comment:
Again, since this is a user-visible message, consider making it a bit nicer; perhaps tell customers what's going on (and why), and even include helpful debug information, like the version.